mongodbreactivestreams/src/main/java/site/ycsb/db/SubscriberHelpers.java

```diff
@@ -0,0 +1,167 @@
+package site.ycsb.db;
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import com.mongodb.MongoTimeoutException;
+import org.bson.Document;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.String.format;
+
+/**
+ *  Subscriber helper implementations for the Quick Tour.
+ */
+public final class SubscriberHelpers {
+
+  /**
+   * A Subscriber that stores the publishers results and provides a latch so can block on completion.
+   *
+   * @param <T> The publishers result type
+   */
+  public static class ObservableSubscriber<T> implements Subscriber<T> {
+    private final List<T> received;
+    private final List<Throwable> errors;
+    private final CountDownLatch latch;
+    private volatile Subscription subscription;
+    private volatile boolean completed;
+
+    ObservableSubscriber() {
+      this.received = new ArrayList<T>();
+      this.errors = new ArrayList<Throwable>();
+      this.latch = new CountDownLatch(1);
+    }
+
+    @Override
+    public void onSubscribe(final Subscription s) {
+      subscription = s;
+    }
+
+    @Override
+    public void onNext(final T t) {
+      received.add(t);
+    }
+
+    @Override
+    public void onError(final Throwable t) {
+      errors.add(t);
+      onComplete();
+    }
+
+    @Override
+    public void onComplete() {
+      completed = true;
+      latch.countDown();
+    }
+
+    public Subscription getSubscription() {
+      return subscription;
+    }
+
+    public List<T> getReceived() {
+      return received;
+    }
+
+    public Throwable getError() {
+      if (errors.size() > 0) {
+        return errors.get(0);
```

Comment:
What do we do with the others?

Reply:
Removed this redundant file SubscriberHelpers.java

Reply:
However, this logic is still there in MongoDbReactiveStreamsClient. In our case mongo stream will be closed in first error itself. So in this case it will always be one error. However, one can use differently and keep collecting the errors and continue on forward with scan result regardless.
Followed the pattern of Mongodb example from their github repo https://github.com/mongodb/mongo-java-driver-reactivestreams/blob/32b31f5f36ffed216c1e21022fed8e5a41cd221e/examples/tour/src/main/tour/SubscriberHelpers.java#L109